pxd_fastpath.c

```diff
@@ -1023,14 +799,23 @@ static int __do_bio_filebacked(struct pxd_device *pxd_dev, struct pxd_io_tracker
 
 	return ret;
 }
-
 #endif
 
-static
-int pxd_handle_io(struct thread_context *tc, struct pxd_io_tracker *head, int dir)
+static void pxd_process_fileio(struct work_struct *wi)
+{
+	struct pxd_io_tracker *iot = container_of(wi, struct pxd_io_tracker, wi);
+	struct pxd_device *pxd_dev = iot->pxd_dev;
+
+	BUG_ON(iot->magic != PXD_IOT_MAGIC);
+	BUG_ON(pxd_dev->magic != PXD_DEV_MAGIC);
+	__do_bio_filebacked(pxd_dev, iot);
+}
+
+static void pxd_process_io(struct pxd_io_tracker *head)
```

Comment:
IO core processing is now direct instead of from a queue earlier. This is the main routine.